src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/CompositionalMultiphaseReservoirHybridFVM.hpp

```diff
@@ -133,10 +137,17 @@ class CompositionalMultiphaseReservoirHybridFVM : public MGRStrategyBase< 4 >
     GEOSX_LAI_CHECK_ERROR( HYPRE_MGRSetLevelSmoothType( precond.ptr, m_levelSmoothType ) );
     GEOSX_LAI_CHECK_ERROR( HYPRE_MGRSetLevelSmoothIters( precond.ptr, m_levelSmoothIters ) );
 
+    // if the wells are shut, using Gaussian elimination as F-relaxation for the well block is an overkill
+    // in that case, we just use Jacobi
+    if( mgrParams.areWellsShut )
+    {
+      m_levelFRelaxType[0] = MGRFRelaxationType::jacobi;
+    }
+
     // Note: uncomment HYPRE_MGRSetLevelFRelaxMethod and comment HYPRE_MGRSetRelaxType breaks the recipe, this requires further
     // investigation
     //GEOSX_LAI_CHECK_ERROR( HYPRE_MGRSetLevelFRelaxMethod( precond.ptr, toUnderlyingPtr( m_levelFRelaxMethod ) ) );
-    GEOSX_LAI_CHECK_ERROR( HYPRE_MGRSetRelaxType( precond.ptr, 0 ));
+    GEOSX_LAI_CHECK_ERROR( HYPRE_MGRSetLevelFRelaxType( precond.ptr, toUnderlyingPtr( m_levelFRelaxType ) ));
```

Comment:
I think this is the intended behavior. In my understanding the goal is to define the f-relaxation exclusively through `HYPRE_MGRSetLevelFRelaxType` (or `HYPRE_MGRSetRelaxType` for a two-level reduction) without specifying whether it is a single- or a multilevel method. Hence, we could completely get rid of `m_levelFRelaxMethod`:
https://github.com/GEOSX/GEOSX/blob/8fb694ec4bda02d1e8691b37310d94622c8be9b0/src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.hpp#L85

@oseikuffuor1 do you agree?

@francoishamon since we have discussed several cleanups for MGR I would address this in another PR.

Reply:
@castelletto1 Yes you are correct. There should be no need to call MGRSetFrelaxMethod anymore.